File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        myList = []
        i = 0
        j = len(numbers) - 1  # Start from the last index
        while i < j:  # Use a loop condition to prevent going beyond the middle
            if numbers[i] + numbers[j] > target:
                j -= 1
            elif numbers[i] + numbers[j] < target:
                i += 1
            else:
                myList.append(i + 1)
                myList.append(j + 1)
                break  # Stop the loop when a solution is found
        return myList

File: 167-two-sum-ii-input-array-is-sorted/TwoSumIIInputArrayIsSorted.java

import java.util.HashMap;
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int n= numbers.length;
        int result[] = new int[2];

        for(int i=0; i<n; i++){
            int complement = target-numbers[i];

            if(map.containsKey(complement)){
                result[0] = map.get(complement) +1;

                result[1] = i+1;

                return result;
            }
            map.put(numbers[i], i);
        }
        return result;
    }
}
